### PR TITLE
Add TrCenterWipeX, TrCenterWipeInX

### PR DESCRIPTION
**Pull request by fdarosa2663**

### CODE REVIEW

File: transitions/center_wipe.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// starts at the POSTION and extends up and down the blade
+// in the specified number of milliseconds.
+template<class MILLIS, class POSITION = Int<16384>>
+class TrCenterWipeX : public TransitionBaseX<MILLIS> {
+public:
+  void run(BladeBase* blade) {
+    pos_.run(blade);
+    TransitionBaseX<MILLIS>::run(blade);
+    center_ = (pos_.getInteger(0) * blade->num_leds()) >> 7;
+    fade_top_ = this->update(256 * blade->num_leds() - center_);
```
Comment (profezzorn) [inline transitions/center_wipe.h:24]: center_, fade_top_ and fade_bottom_ is only used in this function, so they should be local variables.

File: transitions/center_wipe.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  auto getColor(const A& a, const B& b, int led) -> decltype(MixColors(a,b,1,1)) {
+        int mix = (Range(bottom_, top_) & Range(led << 8, (led << 8) + 256)).size();
+    return MixColors(a, b, mix, 8);
+  }
+private:
+  POSITION pos_;
+  uint32_t fade_top_;
+  uint32_t fade_bottom_;
+  uint32_t center_;
+  uint32_t top_;
```
Comment (profezzorn) [inline transitions/center_wipe.h:40]: replace top_ and bottom_ with: Range range_;

File: transitions/center_wipe.h (hunk 1 above)
Comment (fdarosa2663) [reply transitions/center_wipe.h]: fixed

File: transitions/center_wipe.h (hunk 2 above)
Comment (fdarosa2663) [reply transitions/center_wipe.h:40]: I could not get it to compile when trying to replace, how should it be done?

Comment (profezzorn) [reply transitions/center_wipe.h:40]: Replace top_ and bottom_ with:

   Range range_;

Make top and bottom local variables, and then at the end of Run(), do:

   range_ = Range(bottom, top);

in getColor, replace Range(botom, top) with range_.

same below

File: transitions/center_wipe.h (hunk 1 above)
Comment (profezzorn) [reply transitions/center_wipe.h]: did you forget to upload or something? It doesn't look fixed.

Comment (fdarosa2663) [reply transitions/center_wipe.h]: yeah, looks like I pushed a previous, working on the change you suggested but still getting compile error, sending in PM on TRA as it's odd.

File: transitions/center_wipe.h (hunk 2 above)
Comment (fdarosa2663) [reply transitions/center_wipe.h]: Range range_; added and tested, thank you.

**Comment by profezzorn** [comment]
I think this is ready to be merged, but there is a conflict in ProffieOS.ino

**Pull request closed without merging**